src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -47,36 +47,36 @@ public static class PermutationFeatureImportanceExtensions
         /// </format>
         /// </example>
         /// <param name="catalog">The regression catalog.</param>
-        /// <param name="model">The model to evaluate.</param>
+        /// <param name="model">The model on which to evaluate feature importance.</param>
         /// <param name="data">The evaluation data set.</param>
-        /// <param name="label">Label column name.</param>
-        /// <param name="features">Feature column name.</param>
+        /// <param name="labelColumnName">Label column name.</param>
+        /// <param name="featureColumnName">Feature column name.</param>
         /// <param name="useFeatureWeightFilter">Use features weight to pre-filter features.</param>
-        /// <param name="topExamples">Limit the number of examples to evaluate on. null means examples (up to ~ 2 bln) from input will be used.</param>
+        /// <param name="numberOfExamplesToUse">Limit the number of examples to evaluate on. <cref langword="null"/> means up to ~2 bln examples from <paramref param="data"/> will be used.</param>
         /// <param name="permutationCount">The number of permutations to perform.</param>
         /// <returns>Array of per-feature 'contributions' to the score.</returns>
         public static ImmutableArray<RegressionMetricsStatistics>
             PermutationFeatureImportance<TModel>(
                 this RegressionCatalog catalog,
                 IPredictionTransformer<TModel> model,
                 IDataView data,
-                string label = DefaultColumnNames.Label,
-                string features = DefaultColumnNames.Features,
+                string labelColumnName = DefaultColumnNames.Label,
+                string featureColumnName = DefaultColumnNames.Features,
```

Comment:
>string featureColumnName = DefaultColumnNames.Features, [](start = 16, length = 55)

Can't we just use `model.FeatureColumn` here and get rid of this input? Maybe plumb this all we way into the actual `Transformer` to get rid of unnecessary asks in the API?

Reply:
Yes I agree this is a good idea!


---
In reply to: [266087001](https://github.com/dotnet/machinelearning/pull/2966#discussion_r266087001) [](ancestors = 266087001)

Reply:
Actually I disagree, what if you trained your model on some data, and then you want to check the feature importance on another dataset, you will have to name the feature column exactly the same. Isn't that a problem?

---
In reply to: [266142547](https://github.com/dotnet/machinelearning/pull/2966#discussion_r266142547) [](ancestors = 266142547,266087001)

Reply:
You have to use the same input column name to use it as a transformer / scorer anyways, so this is consistent with our schema rules.

---
In reply to: [266150377](https://github.com/dotnet/machinelearning/pull/2966#discussion_r266150377) [](ancestors = 266150377,266142547,266087001)

Reply:
Ok makes sense!

---
In reply to: [266164573](https://github.com/dotnet/machinelearning/pull/2966#discussion_r266164573) [](ancestors = 266164573,266150377,266142547,266087001)